Update release-checklist.md (#1091)
IrisShaders/Iris: docs/development/release-checklist.md

release sufficiently, follow these steps to publish that new release to the public.

1. Change the Iris version in `gradle.properties` on the `trunk`, `1.17`, and `1.18` branch.
2. Push the version bump commit to GitHub.
3. Go to the releases tab on the Iris repository, and click the `Draft a new release` button.
    1. Insert a tag and title relevant to the branch and Iris version
    2. Insert a changelog from the relevant summary file to the Iris version being released
    3. Publish
    4. Repeat for each branch being released

> 4. The action will proceed to run, generate and publish the artifacts necessary, for all release, to CurseForge, GitHub and Modrinth

Reviewer: ```suggestion
4. The action will generate and publish the relevant JAR file to CurseForge, GitHub and Modrinth.
```

Author: Way better than how i put it :P
